roles/ceph-rgw-loadbalancer/tasks/pre_requisite.yml

```diff
@@ -19,7 +19,7 @@
 
 - name: set_fact vip to vrrp_instance
   set_fact:
-    vrrp_instances: "{{ vrrp_instances | default([]) | union([{ 'name': 'VI_' + index|string , 'vip': item }]) }}"
+    vrrp_instances: "{{ vrrp_instances | default([]) | union([{ 'name': 'VI_' + index|string , 'vip': item, 'master': groups[rgwloadbalancer_group_name][index] }]) }}"
```

Comment:
The problem with this is that it doesn't work when you have more VIP than loadbalancer nodes because the iteration is made on the VIP list but you're using the index on the loadbalancer group list.